src/rgw/store/dbstore/sqlite/sqliteDB.cc

```diff
@@ -331,9 +331,12 @@ static int get_objectdata(const DoutPrefixProvider *dpp, DBOpInfo &op, sqlite3_s
   blob = sqlite3_column_blob(stmt, 3);
   datalen = sqlite3_column_bytes(stmt, 3);
 
-  char data[datalen+1] = {};
-  if (blob)
+  char data[datalen+1];
```

Comment:
not quite related, but why are we copying from `blob` and then drop the copied data on the floor?

Reply:
I was wondering the same, but was assuming there was more to follow.

That part of the code already got reviewed/commited

Reply:
@soumyakoduri could you please help on this?

Reply:
hi,
you are right. All *object* related APIs in this file are just placeholders and were to check how to deal with each data type. 

Final version is being worked upon -
https://github.com/soumyakoduri/ceph/blob/dbstore_object/src/rgw/store/dbstore/sqlite/sqliteDB.cc#L452

This particular char array is no more used. These changes will take a while to complete & merged.

Reply:
@soumyakoduri thank you for the explanation.